language-support/hs/bindings/test/DA/Ledger/Tests.hs

```diff
@@ -499,6 +502,75 @@ tAllocateParty withSandbox = testCase "tAllocateParty" $ run withSandbox $ \_pid
     list <- listKnownParties
     liftIO $ assertEqual "list" [expected] list
 
+
+bucket :: Value
+bucket = VRecord $ Record Nothing
+    [ RecordField "record" $ VRecord $ Record Nothing
+        [ RecordField "foo" $ VBool False
+        , RecordField "bar" $ VText "sheep"
+        ]
+    , RecordField "variants" $ VList
+        [ VVariant $ Variant Nothing (ConstructorId "B") (VBool True)
+        , VVariant $ Variant Nothing (ConstructorId "I") (VInt 99)
+        ]
+    , RecordField "contract"$ VContract (ContractId "xxxxx")
+    , RecordField "list"    $ VList []
+    , RecordField "int"     $ VInt 42
+    , RecordField "decimal" $ VDecimal "123.456"
+    , RecordField "text"    $ VText "OMG lol"
+    , RecordField "time"    $ VTime (MicroSecondsSinceEpoch $ 1000 * 1000 * 60 * 60 * 24 * 365 * 50)
+    , RecordField "party"   $ VParty $ Party "good time"
+    , RecordField "bool"    $ VBool False
+    , RecordField "unit"      VUnit
+    , RecordField "date"    $ VDate $ DaysSinceEpoch 123
+    , RecordField "opts"    $ VList
+        [ VOpt Nothing
+        , VOpt $ Just $ VText "something"
+        ]
+    , RecordField "map"     $ VMap $ Map.fromList [("one",VInt 1),("two",VInt 2),("three",VInt 3)]
+    , RecordField "enum"    $ VEnum $ Enum Nothing (ConstructorId "Green")
+
+    ]
+
+tValueConversion :: SandboxTest
+tValueConversion withSandbox = testCase "tValueConversion" $ run withSandbox $ \pid -> do
+    let owner = alice
+    let mod = ModuleName "Valuepedia"
+    let tid = TemplateId (Identifier pid mod $ EntityName "HasBucket")
+    let args = Record Nothing [ RecordField "owner" (VParty owner), RecordField "bucket" bucket ]
+    let command = CreateCommand {tid,args}
+    lid <- getLedgerIdentity
+    _::CommandId <- submitCommand lid alice command >>= either (liftIO . assertFailure) return
+    withGetAllTransactions lid alice (Verbosity True) $ \txs -> do
+    Just elem <- liftIO $ timeout 1 (takeStream txs)
+    trList <- either (liftIO . assertFailure . show) return elem
+    [Transaction{events=[CreatedEvent{createArgs=Record{fields}}]}] <- return trList
+    [RecordField{label="owner"},RecordField{label="bucket",fieldValue=bucketReturned}] <- return fields
+    liftIO $ assertEqual "bucket" bucket (detag bucketReturned)
+
+-- Strip the rid,vid,eid tags recusively from record, variant and enum values
+detag :: Value -> Value
+detag = \case
+    VRecord r -> VRecord $ detagRecord r
+    VVariant v -> VVariant $ detagVariant v
+    VEnum e -> VEnum $ detagEnum e
+    VList xs -> VList $ fmap detag xs
+    VOpt opt -> VOpt $ fmap detag opt
+    VMap m -> VMap $ fmap detag m
+    v -> v
```

Comment:
Instead of having a catch-all case, I would prefer this to be an explicit list of all cases to ensure the compiler catches it when new value constructors are added. But this is test code so maybe it's fine.

Reply:
Agreed.

Reply:
I'm agreeing with 2nd half of Fran's comment - it would be overkill for test code.